spmd/tensor/parallel/style.py

```diff
@@ -27,6 +21,29 @@ class ParallelStyle(ABC):
     _prepare_input: Optional[_Prepare_Input_Func_Type]
     _prepare_output: Optional[_Prepare_Output_Func_Type]
 
+    @abstractmethod
+    def __init__(self, _prepare_input, _prepare_output) -> None:
+        self._prepare_input = _prepare_input
+        self._prepare_output = _prepare_output
+
+
+class PairwiseParallel(ParallelStyle):
+    """
+    We concatenate colwise and rowwise styles as a fixed pair like
+    what Megatron-LM(https://arxiv.org/abs/1909.08053) is doing.
+    We assume both input and output needs to a replicate DTensor.
+    We now only support this style for Multihead Attention, MLP and
+    transformer model.
+
+    .. warning::
+        We need to assume the module to be distributed as a
```

Comment:
Maybe it's not something we "assume", it's something this ParallelStyle requires?